hpx/parallel/algorithms/partition.hpp

```diff
@@ -283,6 +290,300 @@ namespace hpx { namespace parallel { inline namespace v1
                 std::forward<ExPolicy>(policy), is_seq(), first, last,
                 std::forward<F>(f), std::forward<Proj>(proj));
     }
+
+    /////////////////////////////////////////////////////////////////////////////
+    // partition_copy
+    namespace detail
+    {
+        /// \cond NOINTERNAL
+
+        // sequential partition_copy with projection function
+        template <typename InIter, typename OutIter1, typename OutIter2,
+            typename Pred, typename Proj>
+        inline std::pair<OutIter1, OutIter2>
+        sequential_partition_copy(InIter first, InIter last,
+            OutIter1 dest_true, OutIter2 dest_false, Pred && pred, Proj && proj)
+        {
+            while (first != last)
+            {
+                if (hpx::util::invoke(pred, hpx::util::invoke(proj, *first)))
```

Comment:
@hkaiser I guess all required compiler versions should support it:
- GCC since 4.8 or 4.9
- Clang since 3.3 or 3.4
- MSVC since 2015

However, it is safe for the future to support perfect forwarding for callable objects.